src/app/GitUI/UserControls/BlameControl.cs

```diff
@@ -321,14 +321,16 @@ private void BlameFile_VScrollPositionChanged(object sender, EventArgs e)
         private void ProcessBlame(string? filename, GitRevision revision, IReadOnlyList<ObjectId>? children, Control? controlToMask, int lineNumber, CancellationToken cancellationToken = default)
         {
             int avatarSize = BlameAuthor.Font.Height + 1;
-            (string gutter, string body, List<GitBlameEntry> avatars) = BuildBlameContents(filename, avatarSize);
+            (string gutter, string body, List<GitBlameEntry> avatars, int maxLineLength) = BuildBlameContents(filename, avatarSize);
             cancellationToken.ThrowIfCancellationRequested();
 
             BlameAuthor.SetGitBlameGutter(avatars);
 
             Validates.NotNull(_fileName);
 
             BlameAuthor.InvokeAndForget(() => BlameAuthor.ViewTextAsync("committer.txt", gutter));
+            Size size = TextRenderer.MeasureText(new string('a', maxLineLength), BlameAuthor.Font);
+            splitContainer2.SplitterDistance = Math.Max(Math.Min(splitContainer2.SplitterDistance, size.Width + avatarSize), avatarSize);
```

Comment:
`size.Width + avatarSize` should only be taken into account if `splitContainer2.SplitterDistance > splitContainer2.Width / 2`.
Otherwise, the user chosen width can be reduced by e.g. selecting a file without renames.
And `size.Width + avatarSize` should be limited to half `splitContainer2.Width`, too.

Reply:
> After that the user can change the position.

... and gets annoyed because GE thinks it would know better what the perfect position would be. This PR shrinks only.
So, the gutter becomes unreadable each time after a file with short author names and without previous files names was blamed.

If you do not want to use the heuristic with half size, the user chosen width should be remembered and applied if the gutter is wider.

Reply:
I find this slightly annoying too.

> If you do not want to use the heuristic with half size, the user chosen width should be remembered and applied if the gutter is wider.

A solution where the last manual size is used would work better for me.

Maybe file extension should be included, not just the `.`

Reply:
This PR was just a 1st attempt improvement of a better default and not intended to be merged like that. As there is a PR restoring previous value, the added value of the PR is not huge. Maybe one day I will have the energy and time to make it compatible with other PR.
Until that, closing...